### DISCUSSION TITLE
How can i see the the version of linter

### DISCUSSION THREAD

**Discussion by NoSugarCoffee** [golangci/golangci-lint#4049] [Q&A] [answered]
`golangci-lint linters` cmd just print the linter's config but no version.
The scene about this may can help me to know real version of linter then i can do some debug for specific linter

**Comment by atc0005** [comment]
There may be better ways, but this may prove useful until you get a better answer.

For the examples below, assume that you want to determine what version of the `gosec` linter was included.

## Check your installed binary

Run `go version -m $(which golangci-lint)` (or `go version -m $(which golangci-lint).exe`).

```console
$ go version -m $(which golangci-lint).exe | grep gosec
        dep     github.com/securego/gosec/v2    v2.17.0 h1:ZpAStTDKY39insEG9OH6kV3IkhQZPTq9a9eGOLOjcdI=
```

The version of golangci-lint installed on my system includes v2.17.0 of the gosec linter.

## Check go.mod

Each golangic-lint release is associated with a tag. Determine what version you're troubleshooting and consult that tag.

For this example, we're looking at the `v1.54.2` tag. Replace `v1.54.2` with the specific tag that matches the version you are troubleshooting.

Tag links: 

- https://github.com/golangci/golangci-lint/blob/v1.54.2/go.mod
  - detailed UI
- https://raw.githubusercontent.com/golangci/golangci-lint/v1.54.2/go.mod
  - suitable for CLI use

```console
$ curl --silent https://raw.githubusercontent.com/golangci/golangci-lint/v1.54.2/go.mod | grep gosec
        github.com/securego/gosec/v2 v2.17.0
```

------

FWIW: v2.17.0 of the `gosec` linter has a confirmed bug (see https://github.com/securego/gosec/issues/1001).

**Reply by ldez** [reply, marked as answer]
You just have to run `golangci-lint version --debug`

ex:
```console
$ golangci-lint version --debug | grep gosec
dep     github.com/securego/gosec/v2    v2.17.0 h1:ZpAStTDKY39insEG9OH6kV3IkhQZPTq9a9eGOLOjcdI=
```

**Reply by NoSugarCoffee** [reply]
unfortunately, the deug flag not supported at v1.51.0

![image](https://github.com/golangci/golangci-lint/assets/25247325/eec508ba-acb4-4ed5-8cc4-6895b9c4deb4)

Thanks for your answer

**Reply by NoSugarCoffee** [reply]
@atc0005 
It doesn't works for some linters. It also dependend on human to detemine.

For example: I want to detemine staticcheck version in golangci@v1.51.0.  I viewed https://github.com/golangci/golangci-lint/blob/v1.51.0/go.mod but no staticcheck key words. I need to go to `staticcheck` repo then find the module naming. Finally, I got `honnef.co/go/tools v0.4.0-0.dev.0.20221209223220-58c4d7e4b720`, so the version of staticcheck is `v0.4.0-0.dev.0.20221209223220-58c4d7e4b720`

I mean if `golangci-lint linters` can support print more details. But maybe not worth.

Thanks for your help.
